backend/fetch_events.py

```diff
@@ -179,3229 +172,119 @@ def parse_pdf_abstract(pdf_url):
-
-    pagination = page_soup.find('ul', attrs={'class': 'pagination'})
-    if pagination is not None:
-        range_pages = [l.text.replace('Page', '').replace('Current page', '').strip()
-                       for l in pagination.find_all('li')]
-        range_pages = max([int(n_page) for n_page in
-                           range_pages if n_page.isdigit()])
-    else:
-        range_pages = 1
-
-    for n_page in range(range_pages):
-        all_events_url = urljoin(base_url, '/events?page={}'.format(n_page))
-        all_events_soup = BeautifulSoup(requests.get(
-            all_events_url).content, 'html.parser')
-        all_events = all_events_soup.find_all(
-            'div', attrs={'class': 'events-listing'})
-        for event_section in all_events:
-            event_url = urljoin(base_url, event_section.find(
-                'a')['href']) if event_section.find('a') is not None else ''
-            title = event_section.find('h3', attrs={'class': 'events-title'})
-            event_url = urljoin(base_url, title.find('a')['href'] or '')
-            title = title.text.strip() if title is not None else ''
-
-            date = event_section.find('span', attrs={'class': 'news-date'})
-            date = date.text.strip().replace('\nat', '') if date is not None else ''
-            starttime, endtime = find_startend_time(date)
-            if date is not '':
-                date = date.split('\n')[0]
-
-            if event_url is not base_url:
-                event_page = BeautifulSoup(requests.get(
-                    event_url).content, 'html.parser')
-                location = event_page.find('span', attrs={'class': 'metainfo'})
-                if len(location.find_all('span')) >= 2:
-                    location = location.find_all('span')[-1]
-                    location = location.text.replace(
-                        '|', '').strip() if location is not None else ''
-                else:
-                    location = ''
-
-                description = event_page.find_all(
-                    'div', attrs={'class': 'body'})
-                if len(description) >= 2:
-                    description = description[1].get_text()
-                else:
-                    description = ''
-            events.append({
-                'title': title,
-                'date': date,
-                'starttime': starttime,
-                'endtime': endtime,
-                'location': location,
-                'description': description,
-                'url': event_url,
-                'owner': 'Department of Sociology (Sociology)'
-            })
-    return events
-
+    df = df.sort_values('event_index', na_position='last')
+    event_index = df.event_index.iloc[0]
+    r = df.iloc[-1].to_dict()
+    r['event_index'] = event_index
+    return r
 
-def fetch_events_cceb(base_url='https://www.cceb.med.upenn.edu/events'):
+def find_scrapers(module) -> List[Callable]:
```

Comment:
This is very helpful.

@titipata please make sure to review this function.

Reply:
I checked and this works perfectly!